test/spec/controllers/checkout.js

```diff
@@ -0,0 +1,26 @@
+'use strict';
+
+describe('Controller: CheckoutCtrl', function() {
+
+  // load the controller's module
+  beforeEach(module('tmaClientApp'));
+
+  var CheckoutCtrl,
+    rootScope,
+    scope;
+
+  // Initialize the controller and a mock scope
+  beforeEach(inject(function($controller, $rootScope) {
+    rootScope = $rootScope;
+    scope = $rootScope.$new();
+
+    CheckoutCtrl = $controller('CheckoutCtrl', {
+      $scope: scope,
+    });
+  }));
+
+  it('should have basket object on the scope', function() {
+    rootScope.$apply();
+    expect(scope.basket).toBeDefined();
```

Comment:
it would be better to assert that the `basket` is the right thing rather than just `defined`, so something like
```javascript
expect(scope.basket).toEqual(Basket)
```